src/java/org/apache/cassandra/db/marshal/CollectionType.java

```diff
@@ -199,6 +199,15 @@ public boolean isValueCompatibleWithInternal(AbstractType<?> previous)
         return isValueCompatibleWithFrozen(tprev);
     }
 
+    @Override
+    public boolean isSerializationCompatibleWith(AbstractType<?> previous)
+    {
+        if (!isValueCompatibleWith(previous))
+            return false;
+
+        return valueComparator().isSerializationCompatibleWith(((CollectionType)previous).valueComparator());
```

Comment:
```suggestion
        return valueComparator().isSerializationCompatibleWith(((CollectionType<?>)previous).valueComparator());
```

Reply:
There are actually a few other places in `CollectionType` we could get rid of compile warnings (and one in `ReversedType`).

Reply:
:+1: Changed on branch. Thanks.